### ISSUE TITLE
Very vulnerable to a XSS

### ISSUE THREAD

**Issue by TH3AL3X** [Cog-Creators/discord-embed-sandbox#20]
There should be something to sanitize the code, people can run an XSS without problems..

**Comment by Flame442** [comment]
If you have a serious security vulnerability to report, I am reachable on discord in https://discord.gg/red as Flame#2941. This post does not have enough information for me to understand what is going on, so I urge you to send me a private message if you have a legitimate vulnerability to report.

**Comment by TH3AL3X** [comment]
> If you have a serious security vulnerability to report, I am reachable on discord in https://discord.gg/red as Flame#2941. This post does not have enough information for me to understand what is going on, so I urge you to send me a private message if you have a legitimate vulnerability to report.

For example in the inputs an XSS can be executed
![image](https://user-images.githubusercontent.com/13141042/112774236-521b7980-9039-11eb-9999-1791daa356f7.png)

**Comment by NonImportantPerson** [comment]
Yeah but this does not go to any server or even get displayed to anyone else besides you when do that and you can use alert in dev tools so?

**Comment by mcdcam** [comment]
I wouldn't call this a serious or exploitable vulnerability, however, a script injection attack is possible if a user pastes a malicious string into one of the fields. Of course, this would be as good as useless as AFAIK the site doesn't store anything valuable other than the values of the other fields, so there's nothing for an attacker to take.

This is still an issue as entering certain values (eg if you want \<something\> to appear in one of your fields) will make them not show up as they are seen as HTML. (see screenshot)

![image](https://user-images.githubusercontent.com/59149640/128345078-c81bf193-af78-446d-82d4-48a85a018a34.png)

The issue is that there is no input sanitisation performed in the updateEmbed function, for example here:

https://github.com/Cog-Creators/discord-embed-sandbox/blob/c684c6ed1f3e6a0e6d93b73c89700c0223392d29/js/index.js#L44-L45

If the input is first properly sanitised, for example with something like [this](https://stackoverflow.com/a/48226843/11141521), or ideally a proper sanitisation library, the issue would be resolved.

**Comment by djfjfofoc3** [comment]
Considering, afaik, you cannot add URL-parameters for an embed (ie `?title=test&description=<script>alert('hi')</script>&fields=[title=a;value=b;inline=1]`), this doesn't seem like a **Cross-Site**-Scripting Vulnerability, rather a trivial bug.
However, I would still like to see sanitation, so forks of the project which may allow that aren't unknowingly vulnerable to this.
Will likely open a PR that fixes this in a bit.

EDIT: opened a PR that fixes this; see #25
